apps/server/src/modules/lesson/repo/lesson.repo.integration.spec.ts

```diff
@@ -202,35 +202,89 @@ describe('LessonRepo', () => {
 		});
 	});
 
-	describe('updateLessons', () => {
-		it('should update Lessons without deleted user', async () => {
-			// Arrange
-			const userId = new ObjectId();
-			const contentExample: ComponentProperties = {
-				title: 'title',
+	describe('removeUserReference', () => {
+		const setup = async () => {
+			const userId1 = new ObjectId();
+			const userId2 = new ObjectId();
+			const contentExample1: ComponentProperties = {
+				title: 'title1',
 				hidden: false,
-				user: userId,
+				user: userId1,
 				component: ComponentType.TEXT,
-				content: { text: 'test of content' },
+				content: { text: 'test of content1' },
 			};
-			const lesson1 = lessonFactory.buildWithId({ contents: [contentExample, contentExample] });
-			await em.persistAndFlush([lesson1]);
+			const contentExample2: ComponentProperties = {
+				title: 'title2',
+				hidden: false,
+				user: userId2,
+				component: ComponentType.TEXT,
+				content: { text: 'test of content2' },
+			};
+			const lesson1 = lessonFactory.buildWithId({ contents: [contentExample1, contentExample2] });
+			const lesson2 = lessonFactory.buildWithId({ contents: [contentExample1] });
+			const lesson3 = lessonFactory.buildWithId({ contents: [contentExample2] });
+			await em.persistAndFlush([lesson1, lesson2, lesson3]);
 			em.clear();
 
-			// Arrange expected Array after User deletion
-			lesson1.contents[0].user = undefined;
+			return {
+				lesson1,
+				lesson2,
+				lesson3,
+				userId1,
```

Comment:
userId1: userId1.toHexString() from outside the setup we only work with the string format and should represent it in the test as well.